plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/functions/IcebergBucketFunction.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.functions;
+
+import io.airlift.slice.Slice;
+import io.trino.spi.function.Description;
+import io.trino.spi.function.LiteralParameter;
+import io.trino.spi.function.LiteralParameters;
+import io.trino.spi.function.ScalarFunction;
+import io.trino.spi.function.SqlNullable;
+import io.trino.spi.function.SqlType;
+import io.trino.spi.type.Int128;
+import io.trino.spi.type.StandardTypes;
+import org.apache.iceberg.transforms.Transforms;
+import org.apache.iceberg.types.Types;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import static io.trino.spi.type.Decimals.MAX_SHORT_PRECISION;
+import static io.trino.spi.type.StandardTypes.VARCHAR;
+
+public final class IcebergBucketFunction
+{
+    private IcebergBucketFunction() {}
+
+    @Description("A UDF for the Iceberg bucket transform.")
+    @ScalarFunction("iceberg_bucket")
+    @SqlType(StandardTypes.INTEGER)
+    public static long icebergBucket(
+            @SqlNullable @SqlType(VARCHAR) Slice partitionValue,
+            @SqlNullable @SqlType(StandardTypes.INTEGER) Long numberOfBuckets)
```

Comment:
Why is the 2nd argument marked as nullable?

Reply:
Good point. I will update this.